exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/WebSessionResources.java

```diff
@@ -68,16 +68,19 @@ public SocketAddress getRemoteAddress() {
 
   @Override
   public void close() {
-
     try {
       AutoCloseables.close(webUserSession, allocator);
     } catch (Exception ex) {
       logger.error("Failure while closing the session resources", ex);
     }
 
-    // Set the close future associated with this session.
+    // Notify all the listeners of this closeFuture for failure events so that listeners can do cleanup related to this
+    // WebSession. This will be called after every query execution by AnonymousWebUserConnection::cleanupSession and
```

Comment:
If a session is anonymous, is the session tied to a cookie (i.e. a browser?) If not, then does the session end on the close of the connection? If so, would we kill a query as soon as the web connection that submits the query is closed?

We might need a general comment about how this works (if one does not already exist.)

Reply:
From my understanding session is tied to a cookie not a connection. In case of anonymous session, we call close after completion of each query. But for authenticated session the close will be called when HttpSession is invalidated which is during logout or session timeout.